MediaGalleryMetadata/Model/Png/ReadFile.php

```diff
@@ -53,34 +54,20 @@ public function __construct(
         $this->segmentFactory = $segmentFactory;
     }
 
-    /**
-     * @inheritdoc
-     */
-    public function isApplicable(string $path): bool
-    {
-        $resource = $this->driver->fileOpen($path, 'rb');
-        $marker = $this->readHeader($resource);
-        $this->driver->fileClose($resource);
-
-        return $marker == self::PNG_FILE_START;
-    }
-
     /**
      * @inheritdoc
      */
     public function execute(string $path): FileInterface
     {
+        $segments = [];
```

Comment:
I'd prefer to keep variable declaration as close to the usage as possible. (if you'd like to have it at the first line of the function - a private function can be extracted)